pkg/common/data_store.go

```diff
@@ -23,8 +25,33 @@ func OffloadLiteralMap(ctx context.Context, storageClient *storage.DataStore, li
 	if err != nil {
 		return "", errors.NewFlyteAdminErrorf(codes.Internal, "Failed to construct data reference for [%+v] with err: %v", nestedKeys, err)
 	}
-	if err := storageClient.WriteProtobuf(ctx, uri, storage.Options{}, literalMap); err != nil {
+
+	err = Retry(5, async.RetryDelay, func() error {
+		err = storageClient.WriteProtobuf(ctx, uri, storage.Options{}, literalMap)
+		return err
+	})
+
+	if err != nil {
 		return "", errors.NewFlyteAdminErrorf(codes.Internal, "Failed to write protobuf for [%+v] with err: %v", nestedKeys, err)
 	}
+
 	return uri, nil
 }
+
+func Retry(attempts int, delay time.Duration, f func() error) error {
```

Comment:
Ideally we could change and use https://github.com/flyteorg/flyteadmin/blob/master/pkg/async/shared.go#L13 to only retry on certain errors that can be passed as arguments